File: app/src/main/java/com/cameronbanga/skyscraper/services/ATProtoConstants.kt
package com.cameronbanga.skyscraper.services

/**
 * Constants for AT Protocol API limits
 */
object ATProtoLimits {
    object Feed {
        const val MAX_SUGGESTED_FEEDS = 100
        const val MAX_TIMELINE_POSTS = 100
        const val MAX_FEED_POSTS = 100
    }

    object Graph {
        const val MAX_FOLLOWS = 100
        const val MAX_FOLLOWERS = 100
    }

    object Search {
        const val MAX_USERS = 100
        const val MAX_POSTS = 100
    }

    fun clampFeedLimit(limit: Int, max: Int): Int {
        if (limit > max) {
            android.util.Log.w("ATProtoLimits", "Requested limit ($limit) exceeds maximum ($max), clamping to $max")
            return max
        }
        return limit
    }
}

/**
 * Base URLs for AT Protocol services
 */
object ATProtoUrls {
    const val DEFAULT_PDS = "https://bsky.social"
    const val CHAT_API = "https://api.bsky.chat"
    const val PUBLIC_API = "https://public.api.bsky.app"
}

/**
 * AT Protocol collection names
 */
object ATProtoCollections {
    const val POST = "app.bsky.feed.post"
    const val LIKE = "app.bsky.feed.like"
    const val REPOST = "app.bsky.feed.repost"
    const val FOLLOW = "app.bsky.graph.follow"
    const val THREADGATE = "app.bsky.feed.threadgate"
    const val POSTGATE = "app.bsky.feed.postgate"
}

File: Skyscraper-iOS-20251106/Skyscraper/Services/ATProtoConstants.swift
//
//  ATProtoConstants.swift
//  Skyscraper
//
//  AT Protocol API limits and constants
//

import Foundation

/// Constants for AT Protocol API limits
enum ATProtoLimits {
    /// Maximum number of items that can be requested in a single API call
    enum Feed {
        static let maxSuggestedFeeds = 100
        static let maxTimelinePosts = 100
        static let maxFeedPosts = 100
    }

    enum Graph {
        static let maxFollows = 100
        static let maxFollowers = 100
    }

    enum Search {
        static let maxUsers = 100
        static let maxPosts = 100
    }

    /// Clamp a limit value to the maximum allowed
    static func clampFeedLimit(_ limit: Int, max: Int) -> Int {
        if limit > max {
            print("⚠️ Requested limit (\(limit)) exceeds maximum (\(max)), clamping to \(max)")
            return max
        }
        return limit
    }
}
